SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -27,160 +27,69 @@
-  OldAveragePressure         = NULL;
-  RadialEquilibriumPressure  = NULL;
-  ExtAveragePressure         = NULL;
-  AverageDensity             = NULL;
-  OldAverageDensity          = NULL;
-  ExtAverageDensity          = NULL;
-  AverageNu                  = NULL;
-  AverageKine                = NULL;
-  AverageOmega               = NULL;
-  ExtAverageNu               = NULL;
-  ExtAverageKine             = NULL;
-  ExtAverageOmega            = NULL;
-
-
-  /*--- Initialize primitive quantities for turboperformace ---*/
-
-  DensityIn                     = NULL;
-  PressureIn                    = NULL;
-  TurboVelocityIn               = NULL;
-  DensityOut                    = NULL;
-  PressureOut                   = NULL;
-  TurboVelocityOut              = NULL;
-  KineIn                        = NULL;
-  OmegaIn                       = NULL;
-  NuIn                          = NULL;
-  KineOut                       = NULL;
-  OmegaOut                      = NULL;
-  NuOut                         = NULL;
+void CEulerSolver::AeroCoeffsArray::allocate(int size) {
+  _size = size;
+  CD = new su2double[size]; CL = new su2double[size]; CSF = new su2double[size]; CEff = new su2double[size];
+  CFx = new su2double[size]; CFy = new su2double[size]; CFz = new su2double[size]; CMx = new su2double[size];
+  CMy = new su2double[size]; CMz = new su2double[size]; CoPx = new su2double[size]; CoPy = new su2double[size];
+  CoPz = new su2double[size]; CT = new su2double[size]; CQ = new su2double[size]; CMerit = new su2double[size];
+  setZero();
+}
 
+CEulerSolver::AeroCoeffsArray::~AeroCoeffsArray() {
+  delete [] CD; delete [] CL; delete [] CSF; delete [] CEff;
+  delete [] CFx; delete [] CFy; delete [] CFz; delete [] CMx;
+  delete [] CMy; delete [] CMz; delete [] CoPx; delete [] CoPy;
+  delete [] CoPz; delete [] CT; delete [] CQ; delete [] CMerit;
+}
 
-  /*--- Initialize quantities for Giles BC---*/
+void CEulerSolver::AeroCoeffsArray::setZero(int i) {
+  CD[i] = CL[i] = CSF[i] = CEff[i] = 0.0;
+  CFx[i] = CFy[i] = CFz[i] = CMx[i] = 0.0;
+  CMy[i] = CMz[i] = CoPx[i] = CoPy[i] = 0.0;
+  CoPz[i] = CT[i] = CQ[i] = CMerit[i] = 0.0;
+}
 
-  CkInflow                      = NULL;
-  CkOutflow1                    = NULL;
-  CkOutflow2                    = NULL;
+CEulerSolver::CEulerSolver(void) : CSolver() { }
 
-  nodes = nullptr;
-}
+CEulerSolver::CEulerSolver(CGeometry *geometry, CConfig *config,
+                           unsigned short iMesh, const bool navier_stokes) : CSolver() {
 
-CEulerSolver::CEulerSolver(CGeometry *geometry, CConfig *config, unsigned short iMesh) : CSolver() {
+  /*--- Based on the navier_stokes boolean, determine if this constructor is
+   *    being called by itself, or by its derived class CNSSolver. ---*/
+  string description;
+  unsigned short nSecVar;
+  if (navier_stokes) {
+    description = "Navier-Stokes";
+    nSecVar = 8;
+  }
+  else {
+    description = "Euler";
+    nSecVar = 2;
+  }
```

Comment:
...
re using the main CEulerSolver constructor also when instantiating the NS solver, which now only needs to allocate/init its own members.